httpclient5/src/main/java/org/apache/hc/client5/http/impl/DefaultRedirectStrategy.java

```diff
@@ -80,6 +83,36 @@ public boolean isRedirectAllowed(
         return true;
     }
 
+    private boolean isSameAuthority(final HttpHost h1, final HttpHost h2) {
+        if (h1 == null || h2 == null) {
+            return false;
+        }
+        final String host1 = h1.getHostName();
+        final String host2 = h2.getHostName();
+        if (!host1.equalsIgnoreCase(host2)) {
+            return false;
+        }
+        final int port1 = getEffectivePort(h1);
+        final int port2 = getEffectivePort(h2);
+        return port1 == port2;
+    }
+
+    private int getEffectivePort(final HttpHost endpoint) {
```

Comment:
@arturobernalg Please use `SchemePortResolver` to resolve the port of a protocol scheme. Please also use the one created by the builder. See `HttpClientBuilder`